indexers/keyvalue/LevelDBIndexer/__init__.py

```diff
@@ -46,14 +47,39 @@ def get_query_handler(self):
         import plyvel
         return plyvel.DB(self.index_abspath, create_if_missing=True)
 
-    def query(self, key: str, *args, **kwargs) -> Optional['jina_pb2.Document']:
+    def query(self, key: Any) -> Optional[Any]:
         """Find the protobuf chunk/doc using id
 
         :param key: ``id``
         :return: protobuf chunk or protobuf document
         """
-        v = self.query_handler.get(key.encode('utf8'))
+        # key = key.encode('utf8')
+        key = bytes(key)
+        v = self.query_handler.get(key)
         value = None
         if v is not None:
-            value = Parse(json.loads(v.decode('utf8')), jina_pb2.Document())
+            value = Parse(v.decode('utf8'), Document())
         return value
+
+
+    def update(self, keys: Iterator[int], values: Iterator[bytes]):
+        missed = []
+        for key in keys:
+            if self.query_handler.get(bytes(key)) is None:
+                missed.append(key)
+        if missed:
+            # FIXME get indexer name
+            raise KeyError(f'Key(s) {missed} were not found in {self.save_abspath}')
+
+        # hack
+        self.query_handler.close()
+        self.handler_mutex = False
```

Comment:
Is this required for this Indexer?

Reply:
yes, it's required for this indexer as well. The tests are failing if I remove these lines.

Reply:
But I would argue, that is because, we don't set `self.query_handler = self.write_handler` in the initialization. The client itself should be able to handle it.

Reply:
Hah even easier: always use the same handler as Cristian has done with the `MongoDBIndexer`.

Reply:
I removed the lines and just use the query_handler in all crud methods.
However, when I remove the get_add_handler or the get_create_handler, the test fails when saving the index.
`IsADirectoryError: [Errno 21] Is a directory`

Reply:
Keeping it does not harm, does it?

Reply:
Just a bit of extra code